api.planx.uk/helpers.ts

```diff
@@ -213,6 +258,20 @@ const dataMerged = async (
     else ob[nodeId] = node;
   }
 
+  // for every external portal that has been merged, confirm its' latest version was merged. If not, overwrite older snapshot with newest version
+  //   ** this is a final/separate step because older snapshots can be nested in _already_ flattened data (eg not picked up as ComponentType.ExternalPortal above)
```

Comment:
@DafyddLlyr does this comment help clarify what we discussed this morning at all?

Reply:
Yep this is great - suuuper explicit and we'll be able to understand with zero context in 18 months' time 👍 

Thanks for the extra effort in documenting the changes in this PR.